src/avram/migrator/create_trigger_statement.cr

```diff
@@ -0,0 +1,25 @@
+class Avram::Migrator::CreateTriggerStatement
+  def initialize(
+    @table_name : Symbol,
+    @trigger_name : String,
+    @function : String,
+    callback @trigger_when : Symbol = :before,
+    on @trigger_operation : Array(Symbol) = [:update]
```

Comment:
I originally wanted to use Enum instead of Symbol for both of these so we could validate that you only pass `:before` or `:after` and `:insert, :update, :delete`. However, crystal doesn't support using enum inside of an array and passing an array of Symbol. You'd have to pass an Array of that enum which would get pretty gnarly.

Reply:
Makes sense. I think the symbols will work well here